tests/tasks/addons/cwagent.yaml

```diff
@@ -13,25 +13,21 @@ spec:
   - name: region
     default: "us-west-2"
     description: The region where the cluster is in.
-  - name: endpoint
-    default: ""
+  workspaces:
+  - name: config
   steps:
   - name: create-cw-agent
-    image: alpine/k8s:1.22.6
+    image: alpine/k8s:1.23.7
     script: |
       echo "Approving KCM requests"
       kubectl certificate approve $(kubectl get csr | grep "Pending" | awk '{print $1}')  2>/dev/null || true
-      ENDPOINT_FLAG=""
-      if [ -n "$(params.endpoint)" ]; then
-        ENDPOINT_FLAG="--endpoint $(params.endpoint)"
-      fi
-      aws eks $ENDPOINT_FLAG update-kubeconfig --name $(params.cluster-name) --region $(params.region)
+
       #kubectl commands are purely for knowing state of cluster before kicking off the test.
-      kubectl version
-      kubectl config current-context
+      kubectl --kubeconfig=$(workspaces.config.path)/kubeconfig version
```

Comment:
can we export the `kubeconfig` once per task so we don't have to use it many times ?

Reply:
Technically we can..the reason i was passing the kubeconfig explicitly was that it makes it clear which cluster we are interacting with. I guess we interact with both the management cluster and guest in each task. We can have a export of KUBECONFIG post csr approval in each task. wdyt ?

Reply:
Technically we wouldn't need CSR approval either for eks or deveks, it's just for kitctl clusters alone given it doesn't have certificates-controller running(tech-debt). We can go ahead and remove it completely and add it to documentation about the technical debt and workaround for kitctl tasks so we don't have to pollute these tasks anymore.

Reply:
true..we can have an independent approval task for kitctl getting called in its test pipeline as and when needed.

Reply:
have done the modification